kombu/transport/cherami.py

```diff
@@ -0,0 +1,146 @@
+"""Cherami Transport."""
+
+from cherami_client.consumer import Consumer
+from cherami_client.publisher import Publisher
+
+from kombu.async import get_event_loop
+from kombu.log import get_logger
+from kombu.utils.json import loads, dumps
+
+from . import virtual
+
+
+DEFAULT_PORT = 4922
+
+logger = get_logger('kombu.transport.cherami')
+
+class Channel(virtual.Channel):
+    """Cherami Channel."""
+
+    default_message_batch_size = 1  # No. of messages to fetch when get message is called
+
+    _publisher = None
+    _consumer = None
+
+    # message fetching limit
+    prefetch_limit = 0
+    prefetched = 0
+
+    def __init__(self, *args, **kwargs):
+        super(Channel, self).__init__(*args, **kwargs)
+
+        # event_loop
+        self.hub = kwargs.get('hub') or get_event_loop()
+
+        # cherami-client instance kwargs
+        self.kwargs = self.connection.kwargs
+
+        # delivery_tag(Kombu.Transport.QoS) -> cherami_delivery_token(cherami)
+        self.delivery_map = dict()
```

Comment:
Please replace with literal syntax, i.e., `{}`, for justification read https://doughellmann.com/blog/2012/11/12/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/